test/orange/Orange.test.cc

```diff
@@ -1081,6 +1122,18 @@ TEST_F(HexArrayTest, track_out)
     EXPECT_VEC_CLOSE(d2b, refd2b, real_type(1e-5), real_type(1e-5));
 }
 
+// Change direction on a universe boundary to reenter the cell
+TEST_F(TestEM3Test, safety)
+{
+    EXPECT_TRUE(this->params().supports_safety());
+
+    auto geo = this->make_track_view();
+
+    // Initialize in innermost universe
+    geo = Initializer_t{{19.99, 19.9, 19.9}, {0, 1, 0}};
+    EXPECT_SOFT_EQ(0.01, geo.find_safety());
```

Comment:
Just for thoroughness, could you add an additional test point where the universe-1 boundary (`x=19.43`) is closer?